### ISSUE TITLE
Fittings system override material not exported

### ISSUE THREAD

**Issue by CyrilWaechter** [Autodesk/revit-ifc#112]
# revit-Ifc info
Version : Exporter 19.2.0.0 - Alternate UI 19.2.0.0
Revit version : 2019.2

# Issue description
When you export pipes or duct to IFC. SystemType material override is exported only for segments not for fittings.
## Revit 3D view Shaded mode
![image](https://user-images.githubusercontent.com/9622862/58573661-81df7780-823e-11e9-8f82-bfd968040cfc.png)
## Revit System Type settings exemple
![image](https://user-images.githubusercontent.com/9622862/58573793-c10dc880-823e-11e9-94d2-b26d2ee6830f.png)
![image](https://user-images.githubusercontent.com/9622862/58573875-f74b4800-823e-11e9-90f3-74092db7db59.png)
### IFC 2x3 Coordination View 2.0
![image](https://user-images.githubusercontent.com/9622862/58574413-34fca080-8240-11e9-9c8b-83f7a6995ade.png)
### IFC 4 Reference View
![image](https://user-images.githubusercontent.com/9622862/58575375-34fda000-8242-11e9-93e0-52739a35b60b.png)
### IFC 4 Design Transfer View
![image](https://user-images.githubusercontent.com/9622862/58575571-9faedb80-8242-11e9-9ded-84378bb137d5.png)

# Expected Behaviour
IFC 2x3 Coordination View 2.0 : Having fittings exported with SystemType material override as well
IFC 4 Reference View : Having System material override exported
IFC 4 Design Transfer View : Having fittings exported with SystemType material override as well and pipe fitting not dissapearing

Remark : An option to deactivate system override would be great (eg. for rendering purposes) but I don't know if has to be an option in Revit or revit-ifc.

**Comment by ibober** [comment]
The bug is still reproduced.
![image](https://user-images.githubusercontent.com/8260211/121597305-3412c200-ca49-11eb-820d-9f1957d657ad.png)
The model is attached.
[FittingColor.zip](https://github.com/Autodesk/revit-ifc/files/6634413/FittingColor.zip)

**Comment by MJansen22** [comment]
Having the same problem you guys already found a solution fot this?

**Comment by CyrilWaechter** [comment]
@MJansen22 No. You currently need to patch your ifc file after export using [ifcopenshell](https://github.com/IfcOpenShell/IfcOpenShell) / [blenderbim add-on](https://blenderbim.org/) for example.

**Comment by HL-code** [comment]
Tested 2022. Still issue.

![FittingColor](https://user-images.githubusercontent.com/68648654/140488029-4b8c7b6b-260f-4f07-9454-599c1dbaca00.png)

**Comment by tcosse** [comment]
Resolving this would be of much help

**Comment by AndreeHub** [comment]
> @MJansen22 No. You currently need to patch your ifc file after export using [ifcopenshell](https://github.com/IfcOpenShell/IfcOpenShell) / [blenderbim add-on](https://blenderbim.org/) for example.

Did you manage to patch it, i read the recepies on the openshell website and there is nothing conecring colors and ifc fitting. What recepie did you use to patch it?

**Comment by CyrilWaechter** [comment]
> Did you manage to patch it, i read the recepies on the openshell website and there is nothing conecring colors and ifc fitting. What recepie did you use to patch it?

There is currently no recipe for this in official repository. It is possible to create one but no one yet (including myself) seems to be motivated enough to make it.

**Comment by AndreeHub** [comment]
There is a shared parameter in the ifc edporter called IfcSingleMaterialOverride ( https://github.com/Autodesk/revit-ifc/blob/master/Source/RevitIFCTools/IFC%20Shared%20Parameters-RevitIFCBuiltIn_ALL_grouped.txt) 
You can asign this parameter to the fittings, then write in the name of the material for which you wish the color to be added.
This implies that you already have the materials that correspond to the system colors. This is a text parameter and you need to make sure that the name is the same between the value and the Material name in you project. So far its showing good results...
![2022-11-07 14_36_38-BIMcollab ZOOM (free version)_ APO_BT10_XAR1_PP_A_E03_XX_416401#IFC-MODELL_001_V](https://user-images.githubusercontent.com/22151743/200338350-704d6722-343e-4821-9493-106d21c602a9.png)

**Comment by ArtemHnatko** [comment]
The issue is reproducible for IFC 4 Design Transfer View in Revit versions 2022–2026.

Testing environment:
- Autodesk Revit 2022-2026

For IFC 2x3 Coordination View 2.0 and IFC 4 Reference View, fittings are exported with the correct SystemType material override as expected. However, for IFC 4 Design Transfer View, fittings are exported with incorrect material overrides, and pipe fittings are not displayed as intended.

![Image](https://github.com/user-attachments/assets/846a9c0d-3acc-4815-a40e-6f5d75860ea3)

**Comment by ArtemHnatko** [comment]
This issue is also reported in Jira as "REVIT-223253" and is currently in the "Open" status.
